app/commands/v2/patch_link_set.rb

```diff
@@ -18,17 +18,25 @@ def call
         lock_version.save!
 
         grouped_links.each do |group, payload_content_ids|
+          # For each set of links in a LinkSet scoped by link_type, this iterator
+          # deletes the entire existing set and then imports all the links in the
+          # payload, preserving their ordering. The activerecord-import gem is
+          # used to create a single SQL statement for importing all links.
           links = link_set.links.where(link_type: group)
           existing_content_ids = links.pluck(:target_content_id)
 
-          content_ids_to_create = payload_content_ids - existing_content_ids
-          content_ids_to_delete = existing_content_ids - payload_content_ids
+          links.where(target_content_id: existing_content_ids).delete_all
 
-          content_ids_to_create.uniq.each do |content_id|
-            links.create!(target_content_id: content_id)
+          links_to_create = []
+          payload_content_ids.uniq.each_with_index do |content_id, i|
+            links_to_create << Link.new(
+              link_set_id: link_set.id,
+              target_content_id: content_id,
+              link_type: group,
+              position: i
+            )
           end
-
-          links.where(target_content_id: content_ids_to_delete).delete_all
+          Link.import links_to_create
```

Comment:
I don't have any problem with reverting to the standard "one insert per row" if you don't think it will case any noticeable performance issues. Maybe we should benchmark before/after?